File: simulations-java/simulations/moving-man/scala-src/edu/colorado/phet/movingman/ladybug/swing/MyRadioButton.scala
package edu.colorado.phet.movingman.ladybug.swing

import model.Observable
import scala.swing.RadioButton
import java.awt.event.{ActionEvent, ActionListener}

class MyRadioButton(text: String, actionListener: => Unit, getter: => Boolean, addListener: (() => Unit) => Unit) extends RadioButton(text) {
  addListener(update)
  update()
  peer.addActionListener(new ActionListener() {
    def actionPerformed(ae: ActionEvent) = actionListener
  });
  def update() = peer.setSelected(getter)
}

File: branches/test-branch-44224/simulations-java/simulations/density/src/edu/colorado/phet/densityjava/common/MyRadioButton.java
package edu.colorado.phet.densityjava.common;

import javax.swing.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.util.ArrayList;

public class MyRadioButton extends JRadioButton {

    public MyRadioButton(String name, Model model, final Getter<Boolean> getter, final Setter<Boolean> setter) {
        super(name);
        setSelected(getter.getValue());
        addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                setter.setValue(true);
                setSelected(getter.getValue());
            }
        });
        model.addListener(new Unit() {
            public void update() {
                setSelected(getter.getValue());
            }
        });
    }
}
